### PR TITLE
Support billing moz backend

### PR DESCRIPTION
**Pull request by nickpell**
for http://manage.dimagi.com/default.asp?154612#861986

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by nickpell** [comment]
@gcapalbo @biyeun

File: settings.py [hunk 1]
```diff
@@ -1378,3 +1378,5 @@
     'less_debug': LESS_DEBUG,
     'less_watch': LESS_WATCH,
 }
+
+SMS_PREFIXES_BY_COUNTRY = {}
```
Comment (gcapalbo) [inline settings.py]: Don't we need to set the prefixes here in order for this to work?

**Comment by gcapalbo** [comment]
I think one thing that we lose by taking this approach (vs adding a separate prefix column in the fee table) is the ability to have a "catch-all" country code. In other words, right now all numbers that start with 25882 are only able to find an sms fee tied to that prefix, and a catch-all fee for 258 would not apply to 25882 numbers. I think that might easily get lost on someone trying to setup rates for a 258 country code in the future.  How hard would it be to have a separate column on the fee table for the prefix, and then look up by country code + prefix first, then by country code only if nothing is found?

File: settings.py (hunk 1 above)
Comment (nickpell) [inline settings.py]: I'd prefer to do this in salt - https://github.com/dimagi/salt-states/pull/133

**Comment by nickpell** [comment]
@gcapalbo Sounds straightforward - we already return the country code if there is no matching prefix, so all that needs to be added is a fee for the country code

**Comment by gcapalbo** [comment]
I think it would also be good if we can remove the dependency of defining the prefixes in settings.py. If we just add a prefix column to the fee table, then when looking up for country code + prefix we could do something like this:

select * from fee_table where country_code is not null and prefix is not null and <phone_number> like country_code||prefix||'%'

Actually, we could probably even just get rid of country code and have one column called prefix. Then look for <phone_number> like prefix||'%', ordering by the length of prefix descending limit 1.

**Comment by gcapalbo** [comment]
@nickpell Well, even if we added the fee for the country code right now it wouldn't work, because the get_country_code_or_prefix function would always return 25882, for example. So it would never match a fee with just a country code of 258. (See the comment I just made above about a way around this)

**Comment by nickpell** [comment]
closing in favor of https://github.com/dimagi/commcare-hq/pull/5516

**Pull request closed without merging**